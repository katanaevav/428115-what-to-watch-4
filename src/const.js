export const START_URL = `https://4.react.pages.academy`;

export const NO_FILTER = `All genres`;

export const DELAY_BEFORE_START_PREVIEW = 1000;
export const MAX_RENDERED_MOVIES_AT_TIME = 8;

export const MAX_SIMILAR_MOVIES_COUNT = 4;

export const REVIEWS_COLUMNS_COUNT = 2;

export const MIN_VOLUME = 0.0;
export const NO_FULLSCREEN = false;

export const TABS_NAMES = [`Overview`, `Details`, `Reviews`];

export const Screens = {
  MAIN_SCREEN: 0,
  MOVIE_PAGE_SCREEN: 1,
  CINEMA_SCREEN: 2,
  AUTH_SCREEN: 3,
  ADD_REVIEW_SCREEN: 4,
};

export const SavingStatus = {
  SUCCESS: `SUCCESS`,
  FAIL: `FAIL`,
};

export const Tabs = {
  OVERVIEW_TAB: 0,
  DETAILS_TAB: 1,
  REVIEWS_TAB: 2,
};

export const MONTH_NAMES = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
