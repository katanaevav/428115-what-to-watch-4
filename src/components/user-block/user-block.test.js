import React from "react";
import renderer from "react-test-renderer";
import UserBlock from "./user-block.jsx";
import {AuthorizationStatus} from "../../const.js";

it(`Render user block element for not authorized user`, () => {
  const tree = renderer.create(
      <UserBlock
        onOpenAuthScreen={() => {}}
        authorizationStatus={AuthorizationStatus.NO_AUTH}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render user block element for authorized user`, () => {
  const tree = renderer.create(
      <UserBlock
        onOpenAuthScreen={() => {}}
        authorizationStatus={AuthorizationStatus.AUTH}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
