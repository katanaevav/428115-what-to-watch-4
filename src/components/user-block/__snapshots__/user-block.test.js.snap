// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render user block element for authorized user 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="img/avatar.jpg"
      width="63"
    />
  </div>
</div>
`;

exports[`Render user block element for not authorized user 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="#"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
